unified-codes/apps/web/src/components/templates/Details.tsx

```diff
@@ -2,41 +2,107 @@ import * as React from 'react';
 import { Dispatch } from 'redux';
 import { connect } from 'react-redux';
 import { useParams } from 'react-router-dom';
-import { IEntity, } from '@unified-codes/data';
+
+import { IEntity, Property } from '@unified-codes/data';
+import {
+  EntityDetailList,
+  IExternalLink,
+  Grid,
+  Typography,
+  CircularProgress,
+  Backdrop,
+} from '@unified-codes/ui';
+
 import { DetailsActions } from '../../actions';
 import { IState } from '../../types';
+import { ITheme } from '../../muiTheme';
+
+import { withStyles } from '@material-ui/core/styles';
+import { ClassNameMap } from '@material-ui/core/styles/withStyles';
 
 export interface DetailsProps {
- getEntity: (code: string) => void;
- entity?: IEntity
+  getEntity: (code: string) => void;
+  entity?: IEntity;
+  classes: ClassNameMap<any>;
+  isLoading: boolean;
 }
 
+const getStyles = (theme: ITheme) => ({
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+  },
+  root: {
+    paddingLeft: 20,
+    paddingTop: 20
+  }
+});
+
 interface IDetailsParameters {
   code: string;
 }
 
 export type Details = React.FunctionComponent<DetailsProps>;
 
-export const DetailsComponent: Details = ({ entity, getEntity }) => {
+export const DetailsComponent: Details = ({ classes, entity, isLoading, getEntity }) => {
   const { code } = useParams<IDetailsParameters>();
 
   React.useEffect(() => {
     getEntity(code);
   }, []);
```

Comment:
Not very familiar with `useParams`, so apologies if I'm a bit confused by this logic!

Is the `code` coming from the navigation props and being used to then fetch the `entity`, which is then passed back through as a prop?

If you're OK with me refactoring as I go, then I think this stuff should be put into the redux store, thus avoiding the rerenders and stuff. Rather than navigating, would suggest just having this as a `page` component inside a template, and just updating the store with the selected `IEntity`, e.g. `selectedEntity` (on click callback should provide `index` or some other info which can identify this, I think). From that, can just use selectors in `mapStateToProps` to extract all the metadata that is being extracted in the component. 

If we do that, we essentially just need a presentational component which does the rendering bits (not sure whether it's best in the library or web app, though).

Does that make sense?

Reply:
Yeah - it's a param fetched from the react-router (mapped to code via `/:code` in App.tsx). The reason I went with this was so we had the ability to direct link to a specific product page - e.g. I could give you http://localhost:4200/details/c7750265 and you wouldn't need to do a search using the Explorer. 

I see what you're saying though, in that when the code is set (either via react-router or from `selectedEntity` event), it could go into the store rather than being read directly from this template? That could work too - I think.

Reply:
I think can still use navigation, but I think the props should be given to the component directly from the store. I think it's a bit of a code smell to be using `useEffect` for initialising the component (exception can be for data which has to be fetched remotely after the component is rendered).

I think ideally, the component should take the `entity` directly (again comes back to avoiding as much logic in the component as possible). The easiest way to do this is to just have all the `EntityDetails` component state in the redux store (like what I'm doing with explorer). That way when you're designing the components, you can just design them with the props you want, rather than having to tightly couple them to navigation (which `useParams` does, as it makes the component unusable without navigation 😢 ).

Does that make sense? Again, wouldn't worry for now, I'll catch up soon!